### PR TITLE
Namelist array element read

### PR DESCRIPTION
**Pull request by vdonaldson**
A namelist read of an array element requires a descriptor with explicit bounds.

### CODE REVIEW

File: flang/test/Lower/namelist.f90 [hunk 1]
```diff
@@ -46,6 +46,28 @@ program p
...
+  ! CHECK: undefined
+  ! CHECK: fir.address_of
+  ! CHECK: fir.insert_value
+  ! CHECK: fir.shape_shift
+  ! CHECK: fir.embox
+  ! CHECK: fir.insert_value
+  ! CHECK: fir.alloca
+  ! CHECK: fir.undefined
+  ! CHECK: fir.address_of
+  ! CHECK-COUNT-3: fir.insert_value
```
Comment (schweitzpgi) [inline flang/test/Lower/namelist.f90:65]: These check lines are pretty lean.

**Review by schweitzpgi — APPROVED** [review]
Looks like what we discussed. Thanks for the fix.

File: flang/test/Lower/namelist.f90 (hunk 1 above)
Comment (vdonaldson) [reply flang/test/Lower/namelist.f90:65]: Yes; I think the primary ops in the sequence are the main thing to check, absent end-to-end execution results.  I've added a little more detail though.

**Review by jeanPerier — APPROVED** [review]
Looks good to me.

**Pull request merged**